posthog/hogql/ast.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import Any, List, Optional
+
+from pydantic import BaseModel, Extra
+
+
+class AST(BaseModel):
+    class Config:
+        extra = Extra.forbid
+
+
+class Expr(AST):
+    pass
+
+
+class Column(AST):
+    expr: Expr
+    alias: Optional[str] = None
+
+
+class BinaryOperationType(str, Enum):
+    Add = "+"
+    Sub = "-"
+    Mult = "*"
+    Div = "/"
+    Mod = "%"
+
+
+class BinaryOperation(Expr):
+    left: Expr
+    right: Expr
+    op: BinaryOperationType
+
+
+class BooleanOperationType(str, Enum):
+    And = "and"
+    Or = "or"
```

Comment:
I've replaced this with `ast.And` and `ast.Or` in the followup PR. Keeping this 1:1 with the Python AST it's replacing in this PR.